crates/meilisearch/src/routes/indexes/fields.rs

```diff
@@ -0,0 +1,328 @@
+        searchable: FieldSearchConfig { enabled: searchable_enabled },
+        distinct: FieldDistinctConfig { enabled: distinct_enabled },
+        filterable: FieldFilterableConfig {
+            enabled: is_filterable,
+            facet_search: FieldFacetSearchConfig { sort_by },
+            filter: FieldFilterConfig { equality: is_filterable, comparison: is_filterable },
+        },
+        localized: FieldLocalizedConfig { locales },
+    }
+}
+
+pub async fn get_index_fields(
+    index_scheduler: GuardedData<ActionPolicy<{ actions::INDEXES_GET }>, Data<IndexScheduler>>,
+    index_uid: web::Path<String>,
+    params: AwebQueryParameter<ListFields, DeserrQueryParamError>,
+) -> Result<HttpResponse, ResponseError> {
+    let index_uid = meilisearch_types::index_uid::IndexUid::try_from(index_uid.into_inner())?;
+    let index = index_scheduler.index(&index_uid)?;
+    let rtxn = index.read_txn()?;
+
+    let all_field_names: Vec<String> =
+        index.fields_ids_map(&rtxn)?.names().map(|s| s.to_string()).collect();
+
+    let settings = settings(&index, &rtxn, SecretPolicy::RevealSecrets)?;
+
+    let displayed_fields = match &*settings.displayed_attributes {
+        Setting::Set(f) => f.clone(),
+        Setting::Reset | Setting::NotSet => vec!["*".to_string()],
+    };
+    let searchable_fields = match &*settings.searchable_attributes {
+        Setting::Set(f) => f.clone(),
+        Setting::Reset | Setting::NotSet => vec!["*".to_string()],
+    };
+    let distinct_field = match &settings.distinct_attribute {
+        Setting::Set(f) => Some(f),
+        _ => None,
+    };
+    let filterable_rules = match &settings.filterable_attributes {
+        Setting::Set(r) => r.clone(),
+        _ => Vec::new(),
+    };
+    let localized_rules: Vec<LocalizedAttributesRule> = match &settings.localized_attributes {
+        Setting::Set(r) => r.iter().cloned().map(|r| r.into()).collect(),
+        _ => Vec::new(),
+    };
+    let facet_sort_config = match &settings.faceting {
+        Setting::Set(f) => match &f.sort_facet_values_by {
+            Setting::Set(c) => c.clone(),
+            _ => BTreeMap::new(),
+        },
+        _ => BTreeMap::new(),
+    };
+
+    let filtered_fields: Vec<String> = if let Some(search) = &params.search {
+        all_field_names.into_iter().filter(|n| field_matches_search(n, search)).collect()
+    } else {
+        all_field_names
+    };
+
+    let mut enriched_fields: Vec<Field> = filtered_fields
+        .into_iter()
+        .map(|n| {
+            enrich_field(
+                n,
+                &displayed_fields,
+                &searchable_fields,
+                distinct_field,
+                &filterable_rules,
+                &localized_rules,
+                &facet_sort_config,
+            )
+        })
+        .collect();
+
+    if let Some(expr) = &params.filter {
+        let conds = parse_filter(expr);
+        enriched_fields.retain(|f| field_satisfies(f, &conds));
+    }
```

Comment:
Turning the iterator to a vec is a missed opportunity. Use filter on the iterator instead of retain on a vec, you will benefit from `skip` and `limit` from the pagination. The filter will be evaluated lazily and not all fields will be processed. Since filter isn't always set, you may need to use itertools' `Either`